offline/cable_input.F90

```diff
@@ -2887,5 +2924,377 @@ SUBROUTINE allocate_cable_vars(air,bgc,canopy,met,bal,                         &
+
+    ! Reading the start year is easy- just the first 4 characters starting
+    ! from the recorded IndxStartDate.
+    READ(CurrentFile(IndxStartDate:IndxStartDate+3), *)&
+      Dataset%StartYear(FileIndx)
+
+    ! The end year is not as simple as taking the 4 characters at
+    ! IndxEndDate, because the size of the <startdate> string is 3
+    ! characters longer than the YYYYMMDD date specifier. Therefore, shift
+    ! the index back 3 units and then take the next 4 characters
+    READ(CurrentFile(IndxEndDate-3:IndxEndDate), *) Dataset%EndYear(FileIndx)
+
+  END DO BuildDataset
+
+  ! Close the file
+  CLOSE(InputUnit)
+
+  ! Finished the work (we assign the VarNames later). Now delete the temporary
+  ! file we used to store the `ls` output.
+  CALL execute_command_line("rm __FileNameWithNoClashes__.txt")
+END SUBROUTINE prepare_spatiotemporal_dataset
+
+SUBROUTINE open_at_first_file(Dataset)
+  !*## Purpose
+  !
+  ! Set up the ```Dataset``` to retrieve data so that we can avoid having to
+  ! handle start-up processes in the core routines.
+  !
+  !## Method
+  !
+  ! Sets the ```CurrentFileIndx``` to 1 (i.e. the first file in the dataset) and
+  ! associates a unique file ID with the file using NetCDF. Identifies the name
+  ! of the variable used in the NetCDF for the desired variable.
+
+  ! Declare the intent of the Dataset
+  TYPE(SPATIO_TEMPORAL_DATASET), INTENT(INOUT) :: Dataset
+
+  ! Status checker for NF90 operations
+  INTEGER :: ok
+
+  ! Set the current FileIndx to 1, and open that NetCDF file.
+  Dataset%CurrentFileIndx = 1
+
+  ok = NF90_OPEN(Dataset%FileNames(Dataset%CurrentFileIndx), NF90_NOWRITE,&
+    Dataset%CurrentFileID)
+  IF (ok /= NF90_NOERR) THEN
+    CALL handle_err(ok, "Error opening "//Dataset%FileNames&
+      (Dataset%CurrentFileIndx)//" in open_at_first_file.")
+  END IF
+
+  ! Find the NetCDF name associated with the variable
+  CALL find_variable_ID(Dataset)
+END SUBROUTINE open_at_first_file
+
+SUBROUTINE find_variable_ID(Dataset)
+  !*## Purpose
+  !
+  ! Use the list of possible names for the variable to find the NetCDF name
+  ! for the variable.
+  !
+  !## Method
+  !
+  ! Run through a list of possible names for a given variable, and once found,
+  ! save the NetCDF variable ID associated with the variable.
+
+  ! Find the desired variable name in the dataset
+  TYPE(SPATIO_TEMPORAL_DATASET), INTENT(INOUT) :: Dataset
+
+  INTEGER :: ok, VarNameIter
+
+  ! Iterate through our possible variable names to find the relevant name
+  FindVar: DO VarNameIter = 1, SIZE(Dataset%VarNames)
+    ok = NF90_INQ_VARID(Dataset%CurrentFileID, Dataset%VarNames(VarNameIter),&
+      Dataset%CurrentVarID)
+    IF (ok == NF90_NOERR) THEN
+      EXIT FindVar
+    END IF
+  END DO FindVar
+
+  ! Handle the case where we don't find the variable
+  IF (ok /= NF90_NOERR) THEN
+    CALL handle_err(ok, "Error finding desired variable in "//Dataset%FileNames&
+      (Dataset%CurrentFileIndx))
+  END IF
```

Comment:
This was a holdover from the existing code. There is some argument I think to perhaps changing the name of the function to something like `handle_iostatus` because the implication is that the function only handles errors. I think that's a problem for another time, I'll remove these clauses from `IF` blocks.

Reply:
We certainly need to settle on one set of error-handling routines for netcdf and other file I/O. As I said, it is hard for me sometimes to figure out what is new and what is left over from previous code version.